docs/source/mocks.rst

```diff
@@ -1,102 +1,73 @@
 Mocks
 =====
 
-What are they
--------------
+What are mocks?
+---------------
 
-Mocks are deployed contract wrappers that have all of the fake's functionality and even more.
+Mocks are extensions to smart contracts that have all of the functionality of a `fake <./fakes.html>`_ with some extra goodies.
+Behind every mock is a real smart contract (with actual Solidity code!) of your choosing.
+You can **modify the behavior of functions like a fake**, or you can leave the functions alone and calls will pass through to your actual contract code.
+And, with a little bit of smock magic, you can even **modify the value of variables within your contract**! 🥳
 
-Because they are actually deployed contract, they can have actual **logic inside** that can be called through.
-And because they have a storage, **internal variable values can be overwritten** 🥳
+When should I use a mock?
+-------------------------
 
+Generally speaking, mocks are more advanced versions of `fakes <./fakes.html>`_.
+Mocks are most effectively used when you need *some* behavior of a real smart contract but still want the ability to modify things on the fly.
 
-How to use
------------------
+One powerful feature of a mock is that you can modify the value of variables within the smart contract.
+You could, for example, use this feature to test the behavior of a function that changes behavior depending on the value of a variable.
 
-.. container:: code-explanation
+Using mocks
+-----------
 
-  Mocks can be initialized only from deployed contracts. If we would have this contract:
+Initialization
+**************
 
-  .. code-block:: solidity
+Initialize with a contract name
+###############################
 
-    contract Counter {
-      uint256 public count;
+.. code-block:: typescript
 
-      constructor(uint256 _startAt) {
-        count = _startAt;
-      }
+  const myContractFactory = await smock.mock("MyContract");
+  const myContract = await myContractFactory.deploy(...);
```

Comment:
Don't you think it would be useful to add an implementation of types with typechain?

Reply:
Yeah I think that could be useful.